src/test/scala/mesosphere/marathon/integration/UpgradeIntegrationTest.scala

```diff
@@ -0,0 +1,214 @@
+        // lower the memory pressure by limiting threads.
+        "-Dakka.actor.default-dispatcher.fork-join-executor.parallelism-min=2",
+        "-Dakka.actor.default-dispatcher.fork-join-executor.factor=1",
+        "-Dakka.actor.default-dispatcher.fork-join-executor.parallelism-max=4",
+        "-Dscala.concurrent.context.minThreads=2",
+        "-Dscala.concurrent.context.maxThreads=32",
+        s"-DmarathonUUID=$uuid -DtestSuite=$suiteName"
+      ) ++ args
+      Process(cmd, workDir, sys.env.toSeq: _*)
+    }
+  }
+
+  "A simple app" should {
+    "survive an upgrade cycle" in {
+
+      Given("A Marathon 1.4.9 is running")
+      val f = new Fixture()
+      marathon149.start().futureValue
+      (marathon149.client.info.entityJson \ "version").as[String] should be("1.4.9")
+
+      And("a new app in Marathon 1.4.6")
+      val app_146 = f.appProxy(f.testBasePath / "app-149", "v1", instances = 1, healthCheck = None)
+
+      When("The app is deployed")
+      val result = marathon149.client.createAppV2(app_146)
+
+      Then("The app is created")
+      result should be(Created)
+      val tasksBeforeUpgrade = eventually {
+        val tasks = marathon149.client.tasks(app_146.id.toPath).value
+        tasks should have size (1)
+        tasks.foreach { task =>
+          task.state should be("TASK_RUNNING")
+        }
+        tasks
+      }
+
+      marathon149.stop().futureValue
+
+      // Pass upgrade to 1.5.6
+      When("Marathon is upgraded to 1.5.6")
+      marathon156.start().futureValue
+      (marathon156.client.info.entityJson \ "version").as[String] should be("1.5.6")
+
+      Then("All apps from 1.4.9 are still running")
+      val tasksAfterUpgradeTo156 = marathon156.client.tasks(app_146.id.toPath).value
+      tasksAfterUpgradeTo156 should be(tasksBeforeUpgrade)
+
+      marathon156.stop().futureValue
+
+      // Pass upgrade to 1.6.322
+      When("Marathon is upgraded to 1.6.322")
+      marathon16322.start().futureValue
+      (marathon16322.client.info.entityJson \ "version").as[String] should be("1.6.322")
+
+      Then("All apps from 1.4.9 are still running")
+      val tasksAfterUpgradeTo16322 = marathon16322.client.tasks(app_146.id.toPath).value
+      tasksAfterUpgradeTo16322 should be(tasksBeforeUpgrade)
+
+      marathon16322.stop().futureValue
+
+      // Pass upgrade to current
+      When("Marathon is upgraded to the current version")
+      marathonCurrent.start().futureValue
+      (marathonCurrent.client.info.entityJson \ "version").as[String] should be("1.6.0-SNAPSHOT")
+
+      Then("All apps from 1.4.9 are still running")
+      val tasksAfterUpgradeToCurrent = marathonCurrent.client.tasks(app_146.id.toPath).value
+      tasksAfterUpgradeToCurrent should be(tasksBeforeUpgrade)
+
+      marathonCurrent.close()
+      //TODO(karsten): We leak processes.
+    }
+  }
+
+  // TODO(karsten): I'd love to have this test extend MarathonTest but I get NullPointerException for Akka.
```

Comment:
Any help is welcome. I suspect that Akka is created lazily. Thus we get a NullPointerException when we try to access it at the wrong moment.